src/main/java/Parser.java

```diff
@@ -0,0 +1,181 @@
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Parser {
+    public String line = "____________________________________________________________";
+    private Storage storage;
+    private TaskList taskList;
+
+    /**
+     * Constructor for Parser.
+     */
+    public Parser() {
+        storage = new Storage("src/main/java/tasklist.txt");
+        taskList = new TaskList();
+    }
+
+    /**
+     * Greetings to welcome the user.
+     */
+    public void greeting() {
+        System.out.println(line);
+        System.out.println("Hello! I'm Duke\n" +
+                "What can I do for you?");
+        System.out.println(line);
+    }
+
+    /**
+     * Generate the list of tasks.
+     */
+    public void generateList() {
+        System.out.println(line);
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < taskList.getList().size(); i++) {
+            Task t = taskList.getList().get(i);
+            System.out.println((i + 1) + ". " + t.printTask());
+        }
+        System.out.println(line);
+    }
+
+    public void generateSearch(ArrayList<Task> tasks) {
+        System.out.println(line);
+        System.out.println("Here are the matching tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            Task t = tasks.get(i);
+            System.out.println((i + 1) + ". " + t.printTask());
+        }
+        System.out.println(line);
+
+    }
+
+    /**
+     * Parse users' commands and respond to them.
+     * @param s
+     * @throws IOException
+     * @throws IncorrectInputException
+     */
+    public void parse(String s) throws IOException, IncorrectInputException {
```

Comment:
Maybe extract out some parts of this method to shorten it? This method may become even longer as more functions of the application is added later on.